From 6d19bccd9b598d5bf7f077945056c73f1a209c72 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:08:34 -0400
Subject: [usb] option: fix line-control pipe direction

Message-id: <1410887463-20674-443-git-send-email-dzickus@redhat.com>
Patchwork-id: 95264
O-Subject: [RHEL7 PATCH 442/591] USB: option: fix line-control pipe direction
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit dd246f2c8c6be36fac10fc42157797429c6fdbb7
Author: Johan Hovold <jhovold@gmail.com>
Date:   Mon May 26 19:23:11 2014 +0200

    USB: option: fix line-control pipe direction

    The option line-control request has been using the wrong pipe direction,
    while relying on USB core to fix it up.

    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/option.c b/drivers/usb/serial/option.c
index 802c069..2003a66 100644
--- a/drivers/usb/serial/option.c
+++ b/drivers/usb/serial/option.c
@@ -1936,7 +1936,7 @@ static int option_send_setup(struct usb_serial_port *port)
  if (res)
   return res;
 
- res = usb_control_msg(serial->dev, usb_rcvctrlpipe(serial->dev, 0),
+ res = usb_control_msg(serial->dev, usb_sndctrlpipe(serial->dev, 0),
     0x22, 0x21, val, priv->bInterfaceNumber, NULL,
     0, USB_CTRL_SET_TIMEOUT);
 
-- 
1.7.1